core/bodydb.go

```diff
@@ -43,7 +43,8 @@ type BodyDb struct {
 	woCache        *lru.Cache
 	processor      *StateProcessor
 
-	slicesRunning []common.Location
+	slicesRunning   []common.Location
+	processingState bool
```

Comment:
why do we need this bool?

Reply:
Previously we would do a calculation every time to determine if a slice was processingstate, which looked at static attributes of the location variable. The change is a small optimization
https://github.com/dominant-strategies/go-quai/pull/1735/commits/c8db53f887b0cbd274278a0648200f9385099b27